### ISSUE TITLE
Error: current implementaion is limited to 32 live Dim objects at once

### ISSUE THREAD

**Issue by edward-io** [facebookresearch/torchdim#1]
Running cells multiple times results in `current implementaion is limited to 32 live Dim objects at once`

This is probably a KP, but creating an issue for tracking.

To repro in colab:

```
# copied from the colab notebook
!pip3 install  https://download.pytorch.org/whl/nightly/cpu/torch-1.13.0.dev20220614%2Bcpu-cp37-cp37m-linux_x86_64.whl -U
!pip install https://github.com/facebookresearch/torchdim/releases/download/colab/torchdim-1.0-cp37-cp37m-linux_x86_64.whl
import torch
from torchdim import dims
```

Run this in a new cell 5+ times:

```
A = torch.rand(3, 4)
B = torch.rand(4, 5)

i, j, k = dims(3)

r = (A[i, k] * B[k, j]).sum(k)
print(r.order(i, j))
```

Removing the limitation would be useful for playing around with torchdim.

**Comment by zdevito** [comment]
Should be fixed in the colab binaries now. Slightly higher chance of hitting bugs in functorch bindings with the new approach but I agree that trying to use a notebook with this limitation is difficult.
